rfqservice/quote_negotiator.go

```diff
@@ -0,0 +1,77 @@
+package rfqservice
+
+import (
+	"github.com/lightninglabs/taproot-assets/fn"
+	msg "github.com/lightninglabs/taproot-assets/rfqmessages"
+)
+
+// QuoteNegotiator is a struct that handles the negotiation of quotes. It is a
+// subsystem of the RFQ handling system. It determines whether a quote is
+// accepted or rejected.
+type QuoteNegotiator struct {
+	// AcceptedQuotes is a channel which is populated with accepted quotes.
+	AcceptedQuotes *fn.EventReceiver[msg.QuoteAccept]
+
+	// incomingQuoteRequests is a channel which is populated with
+	// unprocessed incoming (received) quote request messages.
+	incomingQuoteRequests <-chan msg.QuoteRequest
+
+	// ErrChan is the handle's error reporting channel.
+	ErrChan <-chan error
+
+	// ContextGuard provides a wait group and main quit channel that can be
+	// used to create guarded contexts.
+	*fn.ContextGuard
+}
+
+// NewQuoteNegotiator creates a new RFQ quote negotiator.
+func NewQuoteNegotiator() (*QuoteNegotiator, error) {
+	acceptedQuotes := fn.NewEventReceiver[msg.QuoteAccept](
+		fn.DefaultQueueSize,
+	)
+
+	return &QuoteNegotiator{
+		AcceptedQuotes: acceptedQuotes,
+
+		ErrChan: make(<-chan error),
+
+		ContextGuard: &fn.ContextGuard{
+			DefaultTimeout: DefaultTimeout,
+			Quit:           make(chan struct{}),
+		},
+	}, nil
+}
+
+// HandleIncomingQuoteRequest handles an incoming quote request.
+func (h *QuoteNegotiator) HandleIncomingQuoteRequest(_ msg.QuoteRequest) error {
+	// TODO(ffranr): Push quote request onto queue. We will need to handle
```

Comment:
So the pattern here would be to have two `select` statements. One for sending the request into `h.incomingQuoteRequests` and one for reading from a result queue. Both `select`s would also read on `h.quit` to make them shutdown when required.
Or we pass in a response channel with the request and wait for an answer there, similar to how we do it in https://github.com/lightninglabs/taproot-assets/blob/d8d50878267a35011c83322ef2d89ee3909a47bb/tapgarden/planter.go#L729.